### PR TITLE
Update Hpo to waffle_hub

### PR DESCRIPTION
**Pull request by kimdanni**
### Test Code
```
from waffle_hub.dataset import Dataset
from waffle_hub.hub import Hub

if __name__ == '__main__':
    n_trials = 10
    sampler_type = "TPESampler"  # You can change this to "TPESampler" or "GridSampler" as needed
    search_space = {
        "lr0": [0.005, 0.05],
        "lrf": [0.001, 0.005],
        "mosaic": [0.6, 1],
        "cos_lr": [True, False],
        "hsv_h" : [0.01, 0.02],
        "hsv_s" : [0.01, 0.02],
        "hsv_v" : [0.01, 0.02],
        "translate" : [0.09, 0.11],
        "scale" : [0.45, 0.55],
        "mosaic" : [0.6, 1]
    }
    hub_name = "test6"
    hub = Hub.new(
        name=hub_name,
        task="classification",
        model_type="yolov8",
        model_size="n",
    )
    print(hub.get_default_train_params(task="CLASSIFICATION", model_type="yolov8", model_size="n"))
    dataset = Dataset.load(name="mnist_classification")
    direction = "maximize"
    hub.hpo(dataset=dataset, n_trials=n_trials, direction=direction, sampler_type=sampler_type, search_space=search_space, epochs=30, image_size=8, device="0")
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -28,3 +28,4 @@ pycocotools>=2.0.2
 py-cpuinfo>=8.0.0
 shapely>=2.0.0
 fire==0.5.0
+optuna
```
Comment (ZeroAct) [inline requirements.txt:31]: please specify version

File: waffle_hub/hub/hub.py
Comment (ZeroAct) [inline waffle_hub/hub/hub.py:1]: make the code work independently along the hpo module.

**Review by echoyunjin — APPROVED** [review]

File: test.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "mosaic": [0.6, 1],
+        "cos_lr": [True, False],
+        "hsv_h": [0.01, 0.02],
+        "hsv_s": [0.01, 0.02],
+        "hsv_v": [0.01, 0.02],
+        "translate": [0.09, 0.11],
+        "scale": [0.45, 0.55],
+        "mosaic": [0.6, 1],
+    }
+    hub_name = "test5"
```
Comment (dustjr1565) [inline test.py:19]: It seems like hardcoding.

File: test.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    dataset = Dataset.load(name="mnist_classification")
+    direction = "maximize"
+    hub.hpo(
+        dataset=dataset,
+        n_trials=n_trials,
+        direction=direction,
+        sampler_type=sampler_type,
+        search_space=search_space,
+        epochs=10,
+        image_size=64,
```
Comment (dustjr1565) [inline test.py:36]: Is there a specific reason for the image size being 64?

**Review by echoyunjin — APPROVED** [review]

**Pull request closed without merging**